stan/math/prim/fun/to_ref.hpp

```diff
@@ -9,6 +9,10 @@ namespace math {
 /**
  * This evaluates expensive Eigen expressions. If given expression involves no
  * calculations this is a no-op that should be optimized away.
+ *
+ * When used to pass the returned value into another function that handles
+ * rvalues and lvalues differently `to_forwarding_ref` should be used instead.
```

Comment:
Does it make sense to do the `to_forwarding_ref` behavior everywhere and just do `to_ref`?

Reply:
What I mean is to make `to_ref` do what `to_forwarding_ref` is doing

Reply:
No. `to_forwarding_ref` is not safe to use in cases such as `const auto& x = to_ref(...)`, as it is returning reference to an object that will be destructed after that line is executed.

Reply:
Actually with removal of forwarding from other places we don't need `to_forwarding_ref `.